pyomo/dae/initialization.py

```diff
@@ -0,0 +1,133 @@
+                    print(index_warning(con.name, index))
+                if not allow_skip:
+                    raise
+                continue
+            if (value(condata.body) - value(condata.upper) > tol or
+                value(condata.lower) - value(condata.body) > tol):
+                inconsistent.add(condata)
+
+    for blk in model.component_objects(Block, active=True):
+        # What if there are time-indexed blocks at multiple levels
+        # of a hierarchy?
+        # My preferred convention is to only check the first (highest-
+        # level) time index, but distinguishing between different-level
+        # time indices is an expensive operation.
+        if not is_explicitly_indexed_by(blk, time):
+            continue
+        if is_in_block_indexed_by(blk, time):
+            continue
+        info = get_index_set_except(blk, time)
+        non_time_set = info['set_except']
+        index_getter = info['index_getter']
+        for non_time_index in non_time_set:
+            index = index_getter(non_time_index, t0)
+            blkdata = blk[index]
+            for condata in blkdata.component_data_objects(Constraint,
+                    active=True):
+                if (value(condata.body) - value(condata.upper) > tol or
+                    value(condata.lower) - value(condata.body) > tol):
+                    if condata in inconsistent:
+                        raise ValueError(
+                            '%s has already been visited. The only way this '
+                            'should happen is if the model has nested time-'
+                            'indexed blocks, which is not supported.')
+                    inconsistent.add(condata)
+
+    return list(inconsistent)
+
+
+def solve_consistent_initial_conditions(model, time, solver):
+    """
+    Solves a model with all Constraints and Blocks deactivated except
+    at the initial value of the Set time. Reactivates Constraints and
+    Blocks that got deactivated.
+
+    Args:
+        model: Model that will be solved
+        time: Set whose initial conditions will remain active for solve
+        solver: Something that implements an solve method that accepts
+                a model as an argument
+
+    Returns:
+        The object returned by the solver's solve method
+    """
+    # Need to deactivate discretization equations, wrt time, at t == 0
+    # This is challenging as the only way (to my knowledge) to do this
+    # is to identify_variables in the expression, find the (assume only one?)
+    # DerivativeVar, and access its get_continuousset_list
+    # I would like a get_continuousset_list for discretization equations.
+    # Possibly as a ComponentMap, possibly as an attribute of some new 
+    # DiscEquation subclass of Constraint
+    # Until I have this, this function will only work for backward
+    # discretization schemes
+
+    # Also, would like to be able to check for zero degrees of freedom here
```

Comment:
Why can't this check be added?

Reply:
The `degrees_of_freedom` I usually use is the IDAES function, so I would have to implement my own here (`number_unfixed_variables_in_activated_equalities - number_activated_equalities`). Also checking for this would be pretty slow. I can implement something like this if you think its worthwhile.

Reply:
I figured that was the case but just wanted to check. It would be a good general utility to have in Pyomo too but it doesn't have to be include in this PR.

Reply:
Also I think it's _slightly_ misleading to call this quantity (`number_unfixed_variables_in_activated_equalities - number_activated_equalities`) "degrees of freedom." If we were to go down the rabbit hole of degree of freedom analysis in Pyomo, I'd feel better if we at least had some way to screen variable-equality incidence matrices for constraints that are guaranteed to be redundant. (I think this is equivalent to performing a symbolic factorization on this incidence matrix, although maybe it could be better accomplished by something like Alex Dowling's LICQ checker that I've heard about.)